composer.json

```
@@ -25,7 +25,7 @@
         "ext-ctype": "*",
         "doctrine/collections": "^1.5",
         "doctrine/common": "^3.3",
-        "doctrine/dbal": "^3.3",
+        "doctrine/dbal": "^3.4.x-dev",
```

Would it be acceptable to bump this version constraint temporarily?

I plan to add a couple more upgrade path improvements to the DBAL 3.4. Since they are not yet released, we will either have to force the release in order to avoid the dependency of the dev ORM on dev DBAL, or we will have to refrain from merging all the changes related to the DBAL 4 until they are implemented in full and the scope of the upgrade path improvements is fully defined.

I believe this change is acceptable since it's unlikely that ORM 3 will be released before DBAL 3.4. Even if this happens, we can release DBAL 3.4 right before that.

---

I think it's fine, as you say we can still force publish 3.4 in the unlikely event we want to publish ORM 3 before it is ready.

---

This is fine, but let's use this constraint instead:

```suggestion
        "doctrine/dbal": "^3.4@dev",
```

---

Yeah, we can use that. How is it better than the original constraint?

---

It's consitent with what we usually use in those situations.